src/com/theironyard/novauc/Main.java

```diff
@@ -1,8 +1,95 @@
 package com.theironyard.novauc;
 
+import spark.Spark;
+import spark.ModelAndView;
+import spark.template.mustache.MustacheTemplateEngine;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
 public class Main {
+    static User user;
+
+
+    static HashMap<String, User> users = new HashMap<>();
+    static ArrayList<String> messages = new ArrayList<String>();
+
 
     public static void main(String[] args) {
-	// write your code here
+        // write your code here
+
+
+        Spark.init();
+
+        Spark.get(
+                "/",
+                ((request, response) -> {
+                    String text = request.queryParams("messageText");
+                    HashMap m = new HashMap();
+                    ArrayList<Message> post = new ArrayList<>();
+                    if (user == null) {
+                        return new ModelAndView(messages, "index.html");
+                    } else {
+                        m.put("name", user.name);
+                        return new ModelAndView(m, "message.html");
+                    }
+                }),
+                new MustacheTemplateEngine()
+        );
+
+        Spark.post(
+                "/login",
+                ((request, response) -> {
+                    String name = request.queryParams("loginName");
+                    user = new User(name);
+                    response.redirect("/");
+                    return "";
+
+                })
+        );
+
+
+
+        Spark.post(
+                "/create-user",
+                ((request, response) -> {
+                    String name = request.queryParams("loginName");
+                    user = new User(name);
+                    response.redirect("/");
+                    return "";
+                })
+        );
+
+        Spark.post(
+                "/create-message",
+                ((request, response) -> {
+                    String text = request.queryParams("messageText");
+                    Message message = new Message(user, text);
+                    message.(messages.add(messages));
+
```

Comment:
Jacob this code isn't the refined code we worked on together.  No need to resubmit.  I know your code for this assignment works as it should because I saw it in action.